opm/simulators/flow/DamarisWriter.cpp

```diff
@@ -117,22 +94,47 @@ int setupWritingPars(Parallel::Communication comm,
     // Set the paramater so that the Damaris servers can allocate the correct amount of memory for the variabe
     // Damaris parameters only support int data types. This will limit models to be under size of 2^32-1 elements
     // ToDo: Do we need to check that local ranks are 0 based ?
-    int dam_err = setParameter("n_elements_local", comm.rank(), elements_rank_sizes[comm.rank()]);
+    int dam_err = setParameter("n_elements_local", elements_rank_sizes[comm.rank()]);
     // Damaris parameters only support int data types. This will limit models to be under size of 2^32-1 elements
     // ToDo: Do we need to check that n_elements_global_max will fit in a C int type (INT_MAX)
-    if( n_elements_global_max <= std::numeric_limits<int>::max() ) {
-        setParameter("n_elements_total", comm.rank(), n_elements_global_max);
+    if ( n_elements_global_max <= std::numeric_limits<int>::max() ) {
+        setParameter("n_elements_total", n_elements_global_max);
     } else {
-        OpmLog::error(fmt::format("( rank:{} ) The size of the global array ({}) is"
-                                  "greater than what a Damaris paramater type supports ({}).  ",
-                                  comm.rank(), n_elements_global_max, std::numeric_limits<int>::max() ));
-        // assert( n_elements_global_max <= std::numeric_limits<int>::max() ) ;
+        if (comm.rank() == 0) {
+            OpmLog::error(fmt::format("The size of the global array ({}) is"
+                                      "greater than what a Damaris paramater type supports ({}).  ",
+                                      n_elements_global_max, std::numeric_limits<int>::max() ));
+        }
         OPM_THROW(std::runtime_error, "setupDamarisWritingPars() n_elements_global_max "
                                       "> std::numeric_limits<int>::max() " + std::to_string(dam_err));
     }
-    
+
     return dam_err;
 }
 
 
+void
+handleError(const int dam_err, Parallel::Communication comm, const std::string& message)
+{
+    // Find if some rank has encountered an error.
+    const bool error = (comm.max(dam_err) != DAMARIS_OK);
```

Comment:
Are we guaranteed to have `DAMARIS_OK` always strictly less than all other error codes for all time?  If not, maybe we could phrase this along the lines of
```C++
const int isOk = dam_err == DAMARIS_OK;
const bool error = comm.sum(isOK) != comm.size();
```
instead?

Reply:
Changed.